### PR TITLE
Create submissions feedback and grading page

### LINKED ISSUE #26: Create Frontend For Grading
- to satisfy assignment 6, we need to create a frontend page for grading, such that when the user submits a paper, we will show feedback
- for now let's assume 2 cases:  when there is no error and when there are errors. For no error case, the page should indicate that the program passes all the test cases and ITS check (perhaps), as we don't have a scoring system yet so there is no need to show what score the student gets. And for the case of errors, we should render their code in an editor, highlight the suggested line of error, and provide feedback accordingly. You can refer to ITS API `POST /cs3213/feedback_error` to see what the response will look like.

### PR DESCRIPTION
**Pull request by vivienherq**
# Pull Request

## Description

Create page to display feedback from ITS for past submissions, currently still a rough skeleton of the page.

## Related Issue(s)

Closes #26

## Screenshots (if applicable)

Screenshot 2024-03-18 at 1 44 09 AM

## Checklist

- [ ] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [ ] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [ ] I will take care of the merging and delete the side-branch after the PR is merged.

## To Do
- [ ] Add assignment question display on the left
- [ ] Allow switching between different questions within assignment
- [x] Display ITS feedback within code editor
- [ ] Refactor page into components for cleaner code

### CODE REVIEW

File: frontend/src/app/(pages)/assignments/submissions/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                  // defaultValue={CODE_SNIPPETS[language]}
+                  onMount={onMount}
+                  value={code}
+                  // onChange={(value) => setValue(value)}
+                />
+              </div>
+              <Spacer y={4} />
+              <div className="row-span-1">
+                <div className="flex w-full flex-col">
+                  <Tabs aria-label="Dynamic tabs" items={tabs}>
```
Comment (tryyang2001) [inline frontend/src/app/(pages)/assignments/submissions/[id]/page.tsx:170]: Perhaps you can change the background color when a tab is on focus, as the default one seems to be too dark:
image

**Review by tryyang2001 — APPROVED** [review]
Overall LGTM. You may want to consider breaking down components such as assignment detail (in the left panel) and code editor renderer (in the right panel) so that the code is not super long within a file.

File: frontend/src/app/(pages)/assignments/submissions/[id]/page.tsx (hunk 1 above)
Comment (vivienherq) [reply frontend/src/app/(pages)/assignments/submissions/[id]/page.tsx]: For now I changed the highlighted colour to blue:
Screenshot 2024-03-18 at 1 44 42 AM

**Comment by vivienherq** [comment]
> Overall LGTM. You may want to consider breaking down components such as assignment detail (in the left panel) and code editor renderer (in the right panel) so that the code is not super long within a file.

Will refactor into components when making changes in future commits!

**Pull request merged**